test/pinning_test.jl

```diff
@@ -66,3 +66,18 @@ end
     cpuids_after = getcpuids()
     @test check_compact_within_numa(cpuids_after)
 end
+
+@testset "Thread Pinning (firstn)" begin
+    pinthreads(sort(getcpuids(); rev=true))
+    @test issorted(getcpuids(); rev=true)
+
+    pinthreads(:firstn)
+    @test issorted(getcpuids())
```

Comment:
Thanks for adding tests for `:firstn`! I suggest you explicitly test for `0:length(getcpuids())-1` here and not only `issorted` because that's what characterises `:firstn`. Note that for `:compact` this would also pass because that's `issorted` as well. (Arguably, on most systems `:compact` and `:firstn` is going to be same anyways, but not on all!)